sql/src/main/java/io/crate/expression/symbol/Function.java

```diff
@@ -75,6 +75,16 @@ public boolean canBeCasted() {
         return info.type() == FunctionInfo.Type.SCALAR || info.type() == FunctionInfo.Type.AGGREGATE;
     }
 
+    @Override
+    public boolean isValueSymbol() {
+        for (Symbol argument : arguments) {
```

Comment:
Is this safe for undeterministic functions with zero arguments?

Reply:
It should be safe because if it is cast, the function but it won't be normalized due to being non-deterministic.